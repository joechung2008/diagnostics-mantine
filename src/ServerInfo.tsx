import { Table } from "@mantine/core";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <Table>
      <Table.Thead>
        <Table.Tr>
          <Table.Th>Name</Table.Th>
          <Table.Th>Value</Table.Th>
        </Table.Tr>
      </Table.Thead>
      <Table.Tbody>
        {items.map((item) => (
          <Table.Tr key={item.name}>
            <Table.Td>{item.name}</Table.Td>
            <Table.Td>{item.value}</Table.Td>
          </Table.Tr>
        ))}
      </Table.Tbody>
    </Table>
  );
};

export default ServerInfo;
