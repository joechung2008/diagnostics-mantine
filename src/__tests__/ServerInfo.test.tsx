import { describe, expect, it } from "vitest";
import ServerInfo from "../ServerInfo";
import { render } from "./test-utils";

describe("ServerInfo", () => {
  it("should render the server info in a table", () => {
    const props: ServerInfoProps = {
      hostname: "host",
      uptime: 123,
      serverId: "id",
      deploymentId: "dep",
      nodeVersions: "v1",
      extensionSync: { totalSyncAllCount: 10 },
    };
    const { container } = render(<ServerInfo {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
