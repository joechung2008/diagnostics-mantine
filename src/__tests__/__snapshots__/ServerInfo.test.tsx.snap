// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > should render the server info in a table 1`] = `
<style
  data-mantine-styles="true"
>
  




</style>
`;
